### ISSUE TITLE
netbird conflicts with adguardhome on ubuntu 22 server

### ISSUE THREAD

**Issue by joshymg** [netbirdio/netbird#1859]
**Describe the problem**

Server already running adguardhome (docker), and everything works. Later I installed netbird, and that makes adguardhome fail on port 53 conflict. 

**To Reproduce**

1. on the adguardhome machine: netbird installed, connected once, then 'netbird down'
2. Self hosted netbird management server, with DNS feature enabled, and nameserver address pointing to the netbird peer IP where I am running adguardhome.
3. 'netbird up' on the adguard host. adguardhome docker exits. Restarting the docker also not working.
4. 'netbird down' on the adguard host.  Starting adguardhome docker succeeds. 

**Expected behavior**

netbird client should not impact adguardhome dns server.

**Are you using NetBird Cloud?**

self-hosted netbird-ce .

**NetBird version**

`netbird 0.27.3`

**Comment by joshymg** [comment]
I think I spoke too soon!!
On a detailed check, it was my custom dns config in netplan config seems to be the cause. 
The nameserver entry was set to my lan router IP instead of 127.0.0.1, while I was testing something else. Reverted it, then 'netplan apply'; all good now. Thanks.

**Comment by lixmal** [comment]
Have you looked into the `--dns-resolver-address` CLI flag or `NB_DNS_RESOLVER_ADDRESS` env var: https://docs.netbird.io/how-to/cli#up

**Comment by joshymg** [comment]
For the time being I have tweaked the Nameserver - Distribution groups, and Access policies, and made it to work. In short, the DNS will not be applied to the adguardhome peer. 
Not sure, if that made it work!!! 
 Will try for a few days and see.

**Comment by bigmehmet** [comment]
> For the time being I have tweaked the Nameserver - Distribution groups, and Access policies, and made it to work. In short, the DNS will not be applied to the adguardhome peer. Not sure, if that made it work!!! Will try for a few days and see.

Thanks for the help,
I added adguard host to a group and disabled dns management for that group,
Now it works,
TY

**Comment by sunjam** [comment]
So this can be closed?

**Comment by nazarewk** [comment]
Hello @joshymg,

We're currently reviewing our open issues and would like to verify if this problem still exists in the [latest NetBird version](https://github.com/netbirdio/netbird/releases).

Could you please confirm if the issue is still there?

We may close this issue temporarily if we don't hear back from you within **2 weeks**, but feel free to reopen it with updated information.

Thanks for your contribution to improving the project!

**Comment by joshymg** [comment]
Yes, this issue is resolved using the distribution group and access policy
work around.

**Comment by jotbed** [comment]
I Just installed Netbird on a Adguard Instance. I had to role back. AdGuard will not start.

**Comment by nazarewk** [comment]
@jotbed @joshymg I might be wrong, but from what I understood, AdGuard container is trying to listen on all possible interfaces on all of the ports (including `53`). Which would make sense for "ease of use" if it were the only thing running on the machine, but is otherwise far from best practice and will interfere with a lot more systems than just NetBird.

I would suggest you to reconfigure AdGuard to listen only on the LAN-facing interfaces and ignore the rest. You can specify by providing `-p <ip>:<host-port>:<container-port>`

**Comment by PizzaLovingNerd** [comment]
Closing as a configuration issue. 

Exclude the AdGuard peer from NetBird DNS management through DNS Settings/distribution groups, or set a custom resolver bind address with `--dns-resolver-address` if another service needs port 53. 

Cheers!
